lib/ex_image_info/types/isobmff.ex

```diff
@@ -0,0 +1,313 @@
+  #       # Mac Preview App - rotate - export as heic generates this box:
+  #       "clli",
+  #       # avif
+  #       "pasp",
+  #       "av1C"
+  #     ]
+  #     defp handle_box(skippable, size, <<rest::binary>>, state)
+  #          when skippable in @skippable do
+  #       case rest do
+  #         <<_::binary-size(size), rest::binary>> ->
+  #           state = state |> State.update_readable_bytes(-size) |> State.update_index()
+  #           parse_box(rest, state)
+  #
+  #         _ ->
+  #           @malformed
+  #       end
+  #     end
+  #
+  # to increase robustness, changing the spec to accept any type, moving to the bottom of `handle_box` functions
+
+  @nestable ["iprp", "ipco"]
+  defp handle_box(nestable, _size, <<rest::binary>>, state)
+       when nestable in @nestable do
+    parse_box(rest, State.update_index(state, 0))
+  end
+
+  defp handle_box("ispe", size, <<rest::binary>>, state) when size >= 12 do
+    case rest do
+      <<_::binary-size(4), width::size(32), height::size(32), _::binary-size(size - 12),
+        rest::binary>> ->
+        state =
+          state
+          |> State.update_readable_bytes(-size)
+          |> State.ispe_box(width, height)
+          |> State.update_index()
+
+        parse_box(rest, state)
+
+      <<_::binary-size(size), _rest::binary>> ->
+        @unimplemented
+
+      _ ->
+        @malformed
+    end
+  end
+
+  defp handle_box("ispe", _size, <<_rest::binary>>, _state), do: @unimplemented
+
+  defp handle_box(
+         "ipma",
+         _size,
+         <<_::binary-size(3), flags3::size(8), entries::size(32), rest::binary>>,
+         state
+       ) do
+    state = State.update_readable_bytes(state, -8)
+    flags3? = Bitwise.band(flags3, 0b00000001) == 1
+
+    {_rest, state} =
+      Enum.reduce(0..(entries - 1)//1, {rest, state}, fn _i, {rest, state} ->
+        read_ipma_entry(rest, flags3?, state)
+      end)
+
+    state
+  end
+
+  defp handle_box("irot", size, <<rotation::size(8), rest::binary>>, state) do
+    state = State.update_readable_bytes(state, -size)
+    rotation = Bitwise.band(rotation, 0x3) * 90
+    state = State.update_index(%{state | rotation: rotation})
+    parse_box(rest, state)
+  end
+
+  defp handle_box("jxlc", _size, <<_rest::binary>>, _state), do: @unimplemented
+
+  defp handle_box(_skippable, size, <<rest::binary>>, state) do
+    case rest do
+      <<_::binary-size(size), rest::binary>> ->
+        state = state |> State.update_readable_bytes(-size) |> State.update_index()
+        parse_box(rest, state)
+
+      _ ->
+        @malformed
+    end
+  end
+
+  defp read_ipma_entry(
+         <<id::size(16), essen_count::size(8), rest::binary>>,
+         flags3?,
+         state
+       ) do
+    state = State.update_readable_bytes(state, -3)
+
+    Enum.reduce(0..(essen_count - 1)//1, {rest, state}, fn _i, {rest, state} ->
+      read_ipma_essen_entry(id, rest, flags3?, state)
+    end)
+  end
+
+  defp read_ipma_essen_entry(
```

Comment:
Future work/Contributions welcomed: no real example reaching this case (`flags?` being true).